jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -705,7 +707,9 @@ public void free()
         @Override
         public String toString()
         {
-            return "%s@%x{%s,%s}".formatted(this.getClass().getSimpleName(), hashCode(), _weak.get(), _strong);
+            ConcurrentEntry<P> weakEntry = _weak.get();
+            ConcurrentEntry<P> strongEntry = _strong;
+            return "%s@%x{w=%s,s=%s,e=%s}".formatted(this.getClass().getSimpleName(), hashCode(), weakEntry == null ? "null" : "non-null", strongEntry == null ? "null" : "non-null", weakEntry);
```

Comment:
Why this "null"/"non-null"?
Just print the weak and strong entries.

Reply:
No, because when an entry is at rest, both the weak and strong refs are non-null, so we'd print the same info twice. This is what we currently do and it makes the dump slightly confusing.

This is why I've changed the string to specify which ref is null or not (the info we need to troubleshoot weak/strong refs) plus the entry itself, to improve clarity.

Reply:
```suggestion
            return "%s@%x{%s,%s}".formatted(this.getClass().getSimpleName(), hashCode(), weakEntry, strongEntry == null ? "free" : "held", weakEntry);
```

Reply:
I don't mind this change. @sbordet would you be happy with it?

Reply:
@lorban I don't like "free"/"held" (never did), so I'd prefer "acquired"/"idle" respectively (I want to know if the entry has been acquired, and "free" does not tell me that).

Also, the suggestion uses 4 parameters, but 5 arguments.

Reply:
@sbordet But the methods are called hold and free. We never agreed on better names so changing them to something else in the dump is just confusing.

The names have to match the methods. Ok to rename methods.. but idle really doesn't convey the same meaning as hold.